hive-runtime/build.gradle

```diff
@@ -43,6 +43,9 @@ project(':iceberg-hive-runtime') {
     }
     implementation(project(':iceberg-nessie')) {
       exclude group: 'com.google.code.findbugs', module: 'jsr305'
+      // Nessie brings Jackson dependencies of version 2.15.0 or later.
+      // These are multi-release jars and doesn't work with Java-8 shadowJar plugin.
+      exclude group: 'com.fasterxml.jackson', module: 'jackson-bom'
```

Comment:
the dependency impact appears to be too broad to me... Is there perhaps some way to alter the declarations on the Nessie side to avoid the exclusions cascade?

Reply:
Do you mean, Nessie can degrade the Jackson and do a minor release and Iceberg can bump to that version?

Reply:
No, I mean whether we can (and I'm not sure it's possible) to adjust Nessie dependency declarations such that Iceberg prefers its own version of Jackson, without repo-wide exclusions. I imagine similar changes may be required in all other downstream projects that build with java 8.

Reply:
The dependencies look like this before the exclude. So, Unless I exclude the `jackson-bom`, the issue still exists for me. 

```
+--- project :iceberg-nessie
|    +--- project :iceberg-api (*)
|    +--- project :iceberg-common (*)
|    +--- project :iceberg-core (*)
|    +--- project :iceberg-bundled-guava
|    +--- org.projectnessie.nessie:nessie-client -> 0.59.0
|    |    +--- com.fasterxml.jackson:jackson-bom:2.15.1 (*)
|    |    +--- com.fasterxml.jackson.core:jackson-core -> 2.15.1 (*)
|    |    +--- com.fasterxml.jackson.core:jackson-databind -> 2.15.1 (*)
|    |    +--- com.fasterxml.jackson.core:jackson-annotations -> 2.15.1 (*)
|    |    +--- org.eclipse.microprofile.openapi:microprofile-openapi-api:3.1
|    |    \--- org.projectnessie.nessie:nessie-model:0.59.0
|    |         +--- com.fasterxml.jackson:jackson-bom:2.15.1 (*)
|    |         +--- com.fasterxml.jackson.core:jackson-databind -> 2.15.1 (*)
|    |         \--- com.fasterxml.jackson.core:jackson-annotations -> 2.15.1 (*)
|    +--- com.fasterxml.jackson.core:jackson-databind -> 2.15.1 (*)
|    \--- com.fasterxml.jackson.core:jackson-core -> 2.15.1 (*)
```

After excluding the `jackson-bom,`  dependencies look like this.
```
+--- project :iceberg-nessie
|    +--- project :iceberg-api (*)
|    +--- project :iceberg-common (*)
|    +--- project :iceberg-core (*)
|    +--- project :iceberg-bundled-guava
|    +--- org.projectnessie.nessie:nessie-client -> 0.59.0
|    |    +--- com.fasterxml.jackson.core:jackson-core -> 2.14.1 (*)
|    |    +--- com.fasterxml.jackson.core:jackson-databind -> 2.14.1 (*)
|    |    +--- com.fasterxml.jackson.core:jackson-annotations -> 2.14.1 (*)
|    |    +--- org.eclipse.microprofile.openapi:microprofile-openapi-api:3.1
|    |    \--- org.projectnessie.nessie:nessie-model:0.59.0
|    |         +--- com.fasterxml.jackson.core:jackson-databind -> 2.14.1 (*)
|    |         \--- com.fasterxml.jackson.core:jackson-annotations -> 2.14.1 (*)
|    +--- com.fasterxml.jackson.core:jackson-databind -> 2.14.1 (*)
|    \--- com.fasterxml.jackson.core:jackson-core -> 2.14.1 (*)
```